backend/src/main/java/com/google/models/Alert.java

```diff
@@ -87,18 +97,34 @@ public Entity toEntity() {
     return alertEntity;
   }
 
+  /** Used when an alert is first created and not converted from an entity. */
+  public static Alert createAlertWithoutId(Timestamp timestampDate, 
```

Comment:
What are your thoughts on methodOverloading for alert creation

ie `createAlert` instead of `createAlertWithoutId` where you won't specify the id in the createAlert argument list?

Reply:
I actually had it as `createAlert` in the start and was debating heavily between which naming to use hahah...
In the end, I decided to go with the `WithoutId` as I thought back to what I read in Effective Java where it says the benefit of using static factory create methods is that you have more control in the naming of these methods compared to a constructor. If I have the name as simply `createAlert`, then it wouldn't make much difference compared to the naming of `Alert()` constructor... 
But I guess the naming part is only one of the many benefits of using static factory create methods, so I am fine with changing it back to `createAlert` if that keeps the name shorter and is clearer for clients using the class :)

Reply:
to clarify: `createAlertWithoutId` is not only used during testing. 

it is used in the actual code where the mock generates these `Alert`s because an ID is only assigned after an `Alert` goes through the following process:
mock -> alerts (no ID yet) -> datastore -> alerts (has ID assigned by datastore)

does this make sense?